lib/openid_token_proxy/token/refresh.rb

```diff
@@ -17,6 +17,13 @@ def require_valid_token
           )
           response.headers['X-Token'] = current_token.access_token
           response.headers['X-Refresh-Token'] = current_token.refresh_token
+
+          token = OpenIDTokenProxy::Token.new(
+            current_token.access_token, current_token.id_token,
+            current_token.refresh_token
+          )
```

Comment:
Actually, I'm pretty sure that doesn't work. The initializer takes three arguments, and it expects the first one to be the access token as a string.